tests/integration/test_engine_integration.py

```diff
@@ -270,3 +270,52 @@ def test_empty_program(self, setup_eng):
 
         state4 = eng.run(p2).state
         assert state3 == state4
+
+    # TODO: when ``shots`` is incorporated into other backends, unmark this test
+    @pytest.mark.backends("gaussian")
+    def test_measurefock_shots(self, setup_eng):
+        """Tests that passing shots with a program containing MeasureFock
+           returns a result whose entries have the right shapes and values"""
+        shots = 5
+        expected = np.zeros(dtype=int, shape=(shots,))
+
+        # all modes
+        eng, p1 = setup_eng(3)
+        with p1.context as q:
+            ops.MeasureFock() | q
+        samples = eng.run(p1, shots=shots).samples
+        assert samples.shape == (shots, 3)
+        assert np.allclose(samples, np.stack([expected]*3, axis=1), atol=0, rtol=0)
+
+        # some modes
+        eng, p2 = setup_eng(3)
+        with p2.context as q:
+            ops.MeasureFock() | (q[0], q[2])
+        samples = eng.run(p2, shots=shots).samples
+        assert samples.shape == (shots, 3)
+        assert np.allclose(samples[:,0].astype(int), expected, atol=0, rtol=0)
+        assert all(s is None for s in samples[:, 1])
+        assert np.allclose(samples[:,2].astype(int), expected, atol=0, rtol=0)
```

Comment:
If the tolerance of this is 0, might as well just make it a standard equality rather than using `np.allclose`.

Reply:
you still need some kind of `all` since an equality comparison will return an array, which `assert` refuses to accept.
I've fixed up now to remove `allclose`, and just use `all` now